### PR TITLE
Add Support for E-Mail

### PR DESCRIPTION
**Pull request by Jbithell**
Support for E-Mail - this is a very early draft of BlueEMail

Still to do:
- [x] Add Composer
- [x] Run Testing
- [x] Add support for Twig (body & templates)
- [x] Attachment support
- [ ] Mail tracking support

### CODE REVIEW

**Comment by Jbithell** [comment]
@vogon101 Please merge if happy, and add in composer stuff

**Comment by vogon101** [comment]
I am going to write a composer setup for BlueUtils

**Comment by Jbithell** [comment]
@vogon101 Major changes made and docs written - This is ready for adding to project, once composer added

**Comment by Jbithell** [comment]
@vogon101 For docs see https://github.com/BluePost/BlueUtils/blob/email/BlueEMail/Readme.md

File: BlueEMail/BlueEmail.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	*/
+	//TODO Require twig and "sendgrid/sendgrid": "~5.1"
+	date_default_timezone_set('UTC');
+	namespace BluePost;
+	class Email {
+		$basicdetails = [];
+		$body = [];
+		private $sendgrid;
+		private $sendgridemail;
+		private $twig; //TODO Set this up - it's currently not initalized!
```
Comment (Jbithell) [inline BlueEMail/BlueEmail.php:43]: @vogon101 Please setup Composer

File: BlueEMail/BlueEmail.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+			if ($plaintextalternative) {
+				$plaincontent = new SendGrid\Content("text/plain", $plaintextalternative);
+	    		$this->sendgridemail->addContent($plaincontent);
+				$this->body['plaintext'] = $plaincontent;
+			}
+    		$htmlcontent = new SendGrid\Content("text/html", $html);
+    		$this->sendgridemail->addContent($htmlcontent);
+			$this->body['html'] = $htmlcontent;
+		}
+		function addattachment() {
```
Comment (Jbithell) [inline BlueEMail/BlueEmail.php]: Add Attachment Support

File: BlueEMail/BlueEmail.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			return false; //Yeah not ready for this yet
+			$attachment = new SendGrid\Attachment();
+			$attachment->setContent("TG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQsIGNvbnNlY3RldHVyIGFkaXBpc2NpbmcgZWxpdC4gQ3JhcyBwdW12");
+			$attachment->setType("application/pdf");
+			$attachment->setFilename("balance_001.pdf");
+			$attachment->setDisposition("attachment");
+			$attachment->setContentId("Balance Sheet");
+			$this->sendgridemail->addAttachment($attachment);
+		}
+		function googleanylitics() {
```
Comment (Jbithell) [inline BlueEMail/BlueEmail.php]: Investigate whether worth adding

File: BlueEMail/Readme.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+### Usage
+```php
+$email = new BluePost\EMail('from@example.com', ["to@example.com"], ["cc@example.com"], ["bcc@example.com"], 'Greetings from BlueUtils BlueEMail');
+$email->body(__DIR__ . "/mytwigfile.twig");
+$email->send();
+```
+### Formatting your twig
+Three example twig files are provided in `/themes/examples/` - one for each theme
+
+In your twig file - the variable `{{basicdetails}}` includes some details of your E-Mail that you may wish to use.
```
Comment (Jbithell) [inline BlueEMail/Readme.md]: @vogon101 Is this too confusing?

**Comment by Jbithell** [comment]
Attachment support now completed

File: BlueEMail/BlueEmail.php (hunk 2 above)
Comment (Jbithell) [reply BlueEMail/BlueEmail.php]: Completed

**Comment by vogon101** [comment]
Looks good, I will add composer support and make some changes to the class's API.

I am also considering some sort of universal BlueUtils settings something for things like API keys etc...

**Pull request merged**